tests/flytekit/unit/models/test_execution.py

```diff
@@ -15,6 +18,35 @@
 )
 
 
+def test_execution_closure():
+    test_datetime = datetime.datetime(year=2022, month=1, day=1, tzinfo=pytz.UTC)
+    test_timedelta = datetime.timedelta(seconds=10)
+    test_error = _core_exec.ExecutionError(
+        code="foo", message="bar", error_uri="http://foobar", kind=_core_exec.ExecutionError.ErrorKind.USER
+    )
+    test_outputs = _execution.LiteralMapBlob(values=_OUTPUT_MAP, uri="http://foo/")
+
+    obj = _execution.ExecutionClosure(
+        phase=_core_exec.WorkflowExecutionPhase.SUCCEEDED,
+        started_at=test_datetime,
+        duration=test_timedelta,
+        error=test_error,
+        outputs=test_outputs,
+    )
+    assert obj.phase == _core_exec.WorkflowExecutionPhase.SUCCEEDED
+    assert obj.started_at == test_datetime
+    assert obj.duration == test_timedelta
+    assert obj.error == test_error
+    assert obj.outputs == test_outputs
+    obj2 = _execution.ExecutionClosure.from_flyte_idl(obj.to_flyte_idl())
+    assert obj2 == obj
+    assert obj2.phase == _core_exec.WorkflowExecutionPhase.SUCCEEDED
+    assert obj2.started_at == test_datetime
+    assert obj2.duration == test_timedelta
+    # assert obj2.error == test_error  # FIXME: This won't work for some reason?
```

Comment:
what error are you seeing here?

Reply:
I believe the issue is that after serializing and deserializing the `obj` the error field goes blank
![image](https://user-images.githubusercontent.com/4410453/151639294-89603a97-017b-475a-9d97-46330b97cc30.png)

Reply:
I did some digging and this is a separate bug. If you check the [protobuf definition](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/admin/execution.proto#L127-L144), `ExecutionClosure` has a `one_of` field that should cover `outputs` and `error`, i.e. those two fields should not be set at the same time. Unfortunately the protobuf python library [does not raise any error in this case](https://github.com/flyteorg/flytekit/blob/master/flytekit/models/execution.py#L341-L345) (essentially the last write wins, in this case `outputs`). 

tldr: this is a 3 year old bug in the use of the python protobuf library and should not impact your change.